ConCurrentModificationException running dbm-gorm-diff in Grails 3.0.11

I can't seem to run within a multi-module gradle project the 'dbm-gorm-diff' command:

Command execution error: liquibase.command.CommandExecutionException: java.util.ConcurrentModificationException
dbmGormDiff FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dbmGormDiff'.
> Process 'command '/usr/lib/jvm/java-7-oracle/bin/java'' finished with non-zero exit value 1

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':dbmGormDiff'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
	at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
	at org.gradle.internal.Factories$1.create(Factories.java:22)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
	at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
	at org.gradle.tooling.internal.provider.runner.BuildModelActionRunner.run(BuildModelActionRunner.java:46)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.tooling.internal.provider.runner.SubscribableBuildActionRunner.run(SubscribableBuildActionRunner.java:58)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:77)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:47)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:52)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
	at org.gradle.util.Swapper.swap(Swapper.java:38)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.DaemonHealthTracker.execute(DaemonHealthTracker.java:47)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:66)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.HintGCAfterBuild.execute(HintGCAfterBuild.java:41)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:246)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: org.gradle.process.internal.ExecException: Process 'command '/usr/lib/jvm/java-7-oracle/bin/java'' finished with non-zero exit value 1
	at org.gradle.process.internal.DefaultExecHandle$ExecResultImpl.assertNormalExitValue(DefaultExecHandle.java:367)
	at org.gradle.process.internal.DefaultJavaExecAction.execute(DefaultJavaExecAction.java:31)
	at org.gradle.api.tasks.JavaExec.exec(JavaExec.java:75)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:227)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:220)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:209)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	... 70 more

--- a/build.gradle
+++ b/build.gradle
@@ -17,7 +17,7 @@ plugins {
     id "com.jfrog.bintray" version "1.1"
 }
 
-version "2.0.0.RC1"
+version "2.0.0.RC2"
 group "org.grails.plugins"
 
 apply plugin: "spring-boot"
@@ -53,7 +53,8 @@ dependencyManagement {
 }
 
 dependencies {
-    compile 'org.liquibase:liquibase-core:3.3.2'
+    compile 'org.liquibase:liquibase-core:3.4.2'
+
     compile('org.liquibase.ext:liquibase-hibernate4:3.5') {
         exclude group: 'org.hibernate', module: 'hibernate-core'
         exclude group: 'org.hibernate.javax.persistence', module: 'hibernate-jpa-2.0-api'

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,4 +1,4 @@
-grailsVersion=3.0.1
+grailsVersion=3.0.11
 gradleWrapperVersion=2.3
 
 websiteUrl=http://grails-plugins.github.io/grails-database-migration

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Tue Feb 17 00:01:17 JST 2015
+#Wed Mar 09 10:35:09 GMT 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.3-all.zip

--- a/settings.gradle
+++ b/settings.gradle
@@ -1,0 +1,1 @@
+rootProject.name = "database-migration"

--- a/src/integration-test/groovy/org/grails/plugins/databasemigration/AutoRunWithSingleDataSourceSpec.groovy
+++ b/src/integration-test/groovy/org/grails/plugins/databasemigration/AutoRunWithSingleDataSourceSpec.groovy
@@ -18,6 +18,7 @@ package org.grails.plugins.databasemigration
 import grails.test.mixin.integration.Integration
 import groovy.sql.Sql
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Primary
 import org.springframework.test.context.ActiveProfiles
 import spock.lang.AutoCleanup
 import spock.lang.Specification

--- a/src/main/groovy/org/grails/plugins/databasemigration/command/ApplicationContextDatabaseMigrationCommand.groovy
+++ b/src/main/groovy/org/grails/plugins/databasemigration/command/ApplicationContextDatabaseMigrationCommand.groovy
@@ -66,7 +66,7 @@ trait ApplicationContextDatabaseMigrationCommand implements DatabaseMigrationCom
     }
 
     @CompileDynamic
-    private Database createGormDatabase(ConfigurableApplicationContext applicationContext, String dataSource = null) {
+    private Database createGormDatabase(ConfigurableApplicationContext applicationContext, String dataSource) {
         String sessionFactoryName = dataSource ? '&sessionFactory_' + dataSource : '&sessionFactory'
 
         def sessionFactory = applicationContext.getBean(sessionFactoryName)

--- a/src/main/groovy/org/grails/plugins/databasemigration/command/DbmCreateChangelog.groovy
+++ b/src/main/groovy/org/grails/plugins/databasemigration/command/DbmCreateChangelog.groovy
@@ -16,6 +16,7 @@
 package org.grails.plugins.databasemigration.command
 
 import groovy.transform.CompileStatic
+import liquibase.serializer.ChangeLogSerializer
 import liquibase.serializer.ChangeLogSerializerFactory
 import org.grails.plugins.databasemigration.DatabaseMigrationException
 
@@ -41,10 +42,10 @@ class DbmCreateChangelog implements ScriptDatabaseMigrationCommand {
             outputChangeLogFile.parentFile.mkdirs()
         }
 
-        def serializer = ChangeLogSerializerFactory.instance.getSerializer(filename)
+        ChangeLogSerializer serializer = ChangeLogSerializerFactory.instance.getSerializer(filename)
 
         outputChangeLogFile.withOutputStream { OutputStream out ->
-            serializer.write([], out)
+            serializer.write(new ArrayList(), out)
         }
 
         if (hasOption('add')) {

--- a/src/main/groovy/org/grails/plugins/databasemigration/liquibase/GrailsLiquibase.groovy
+++ b/src/main/groovy/org/grails/plugins/databasemigration/liquibase/GrailsLiquibase.groovy
@@ -23,6 +23,7 @@ import liquibase.exception.LiquibaseException
 import liquibase.integration.spring.SpringLiquibase
 import liquibase.resource.ClassLoaderResourceAccessor
 import liquibase.resource.CompositeResourceAccessor
+import liquibase.resource.ResourceAccessor
 import org.springframework.context.ApplicationContext
 
 import java.sql.Connection
@@ -42,7 +43,8 @@ class GrailsLiquibase extends SpringLiquibase {
 
     @Override
     protected Liquibase createLiquibase(Connection connection) throws LiquibaseException {
-        Liquibase liquibase = new Liquibase(getChangeLog(), new ClassLoaderResourceAccessor(), createDatabase(connection))
+        Liquibase liquibase = new Liquibase(getChangeLog(), new ClassLoaderResourceAccessor(), createDatabase
+                (connection, null))
         liquibase.setIgnoreClasspathPrefix(isIgnoreClasspathPrefix())
         if (parameters != null) {
             for (Map.Entry<String, String> entry : parameters.entrySet()) {
@@ -57,9 +59,10 @@ class GrailsLiquibase extends SpringLiquibase {
         return liquibase
     }
 
+
     @Override
-    protected Database createDatabase(Connection connection) throws DatabaseException {
-        Database database = super.createDatabase(connection)
+    protected Database createDatabase(Connection connection, ResourceAccessor accessor) throws DatabaseException {
+        Database database = super.createDatabase(connection, accessor)
 
         if (databaseChangeLogTableName) {
             database.databaseChangeLogTableName = databaseChangeLogTableName

--- a/src/main/groovy/org/grails/plugins/databasemigration/liquibase/GroovyChangeLogParser.groovy
+++ b/src/main/groovy/org/grails/plugins/databasemigration/liquibase/GroovyChangeLogParser.groovy
@@ -92,7 +92,7 @@ class GroovyChangeLogParser extends AbstractChangeLogParser {
                 databases = value.databases
                 value = value.value
             }
-            changeLogParameters.set(name as String, value as String, contexts as String, labels, databases)
+            changeLogParameters.set(name as String, value as String, contexts as String, labels, databases, true, null)
         }
     }
 }

--- a/src/main/groovy/org/grails/plugins/databasemigration/liquibase/GroovyChangeLogSerializer.groovy
+++ b/src/main/groovy/org/grails/plugins/databasemigration/liquibase/GroovyChangeLogSerializer.groovy
@@ -16,6 +16,7 @@
 package org.grails.plugins.databasemigration.liquibase
 
 import groovy.transform.CompileStatic
+import liquibase.changelog.ChangeLogChild
 import liquibase.changelog.ChangeSet
 import liquibase.serializer.ChangeLogSerializer
 import liquibase.serializer.LiquibaseSerializable
@@ -27,9 +28,9 @@ class GroovyChangeLogSerializer implements ChangeLogSerializer {
     private XMLChangeLogSerializer xmlChangeLogSerializer = new XMLChangeLogSerializer()
 
     @Override
-    void write(List<ChangeSet> changeSets, OutputStream out) throws IOException {
+    def <T extends ChangeLogChild> void write(List<T> changesets, OutputStream out) throws IOException {
         def xmlOutputStrem = new ByteArrayOutputStream()
-        xmlChangeLogSerializer.write(changeSets, xmlOutputStrem)
+        xmlChangeLogSerializer.write(changesets, xmlOutputStrem)
         out << ChangelogXml2Groovy.convert(xmlOutputStrem.toString())
     }
 
@@ -46,5 +47,10 @@ class GroovyChangeLogSerializer implements ChangeLogSerializer {
     @Override
     String serialize(LiquibaseSerializable object, boolean pretty) {
         throw new UnsupportedOperationException()
+    }
+
+    @Override
+    int getPriority() {
+        return 0
     }
 }

